src/filler/handlers/blends/processors/superblends.ts

```diff
@@ -160,26 +141,109 @@ const superBlendsListener = (core: CollectionsListHandler, contract: string) =>
     }
 };
 
+const superBlendsRollsListener = (core: CollectionsListHandler, contract: string) => async (db: ContractDBTransaction, block: ShipBlock, tx: EosioTransaction, trace: EosioActionTrace<SetBlendRollsActionData>): Promise<void> => {
+    const {
+        rollsDbRows,
+        rollOutcomesDbRows,
+        rollOutcomeResultsDbRows,
+    } = getRollsDbRows(
+        trace.act.data.blend_id,
+        trace.act.data.rolls,
+        core.args, block.block_num,
+        block.timestamp,
+        contract,
+    );
+    await deleteBlendRolls(
+        db,
+        core.args.atomicassets_account,
+        trace.act.data.blend_id,
+        contract,
+    );
+    await insertBlendRolls(
+        db,
+        rollsDbRows,
+        rollOutcomesDbRows,
+        rollOutcomeResultsDbRows,
+    );
+};
+
 export function superBlendsProcessor(core: CollectionsListHandler, processor: DataProcessor): () => any {
     const destructors: Array<() => any> = [];
     const neftyContract = core.args.nefty_blender_account;
     const tagContract = core.args.tag_blender_account;
 
     destructors.push(processor.onContractRow(
         neftyContract, 'blends',
-        superBlendsListener(core, neftyContract),
-        BlendsUpdatePriority.TABLE_FEATURES.valueOf()
+        superBlendsTableListener(core, neftyContract),
+        BlendsUpdatePriority.TABLE_BLENDS.valueOf()
+    ));
+
+    destructors.push(processor.onActionTrace(
+        neftyContract, 'setrolls',
+        superBlendsRollsListener(core, neftyContract),
+        BlendsUpdatePriority.SET_ROLLS.valueOf()
     ));
 
     destructors.push(processor.onContractRow(
         tagContract, 'blends',
-        superBlendsListener(core, tagContract),
-        BlendsUpdatePriority.TABLE_FEATURES.valueOf()
+        superBlendsTableListener(core, tagContract),
+        BlendsUpdatePriority.TABLE_BLENDS.valueOf()
+    ));
+
+    destructors.push(processor.onActionTrace(
```

Comment:
why do we need to add a processor to the destructor? I'm just asking to know what's the difference between this and when I had done one to set a property (think it was the blend visibility)?

Reply:
I decided to use this because the table is updated very frequently. The rolls are not. So if we update the rolls every time we update the table in the smart contract we are wasting many resources.